File: src/main/java/com/graqr/threshr/model/redsky/products/OrderPickup.java
package com.graqr.threshr.model.redsky.products;

import com.fasterxml.jackson.annotation.JsonProperty;
import io.micronaut.core.annotation.Nullable;
import io.micronaut.serde.annotation.Serdeable;

@Serdeable
public record OrderPickup(
        @JsonProperty("availability_status")
        String availabilityStatus,
        @Nullable
        @JsonProperty("pickup_date")
        String pickupDate,
        @Nullable
        @JsonProperty("guest_pick_sla")
        Long guestPickSla,
        @Nullable
        @JsonProperty("location_locale")
        String locationLocale) {
}

File: src/main/kotlin/com/graqr/threshr/model/redsky/product/OrderPickup.kt
package com.graqr.threshr.model.redsky.product

import com.fasterxml.jackson.annotation.JsonProperty
import io.micronaut.core.annotation.Nullable
import io.micronaut.serde.annotation.Serdeable

@Serdeable
data class OrderPickup(
    @param:JsonProperty("availability_status") val availabilityStatus: String,
    @param:Nullable @param:JsonProperty("pickup_date") val pickupDate: String,
    @param:Nullable @param:JsonProperty("guest_pick_sla") val guestPickSla: Long,
    @param:Nullable @param:JsonProperty("location_locale") val locationLocale: String
) {}
